x-pack/plugins/ml/public/application/model_management/deployment_setup.tsx

```diff
@@ -63,300 +129,494 @@ export const DeploymentSetup: FC<DeploymentSetupProps> = ({
+              min={0.5}
+              max={2.5}
+              step={1}
+              tickInterval={1}
+              value={vCpuLevelMap[config.vCPUUsage].value}
+              onChange={(e) => {
+                const result = Object.entries(vCpuLevelMap).find(
+                  ([, val]) => val.value === Number(e.currentTarget.value)
+                );
+                onConfigChange({
+                  ...config,
+                  vCPUUsage: result![0]! as DeploymentParamsUI['vCPUUsage'],
+                });
               }}
-              options={threadsPerAllocationsOptions}
-              data-test-subj={'mlModelsStartDeploymentModalThreadsPerAllocation'}
+              showTicks
+              ticks={customTicks}
+              levels={customColorsLevels}
+              aria-label={
+                showNodeInfo
+                  ? i18n.translate('xpack.ml.trainedModels.modelsList.startDeployment.vCpuLevel', {
+                      defaultMessage: 'vCPUs level selector',
+                    })
+                  : i18n.translate('xpack.ml.trainedModels.modelsList.startDeployment.vCULevel', {
+                      defaultMessage: 'VCUs level selector',
+                    })
+              }
+              aria-describedby={'vCpuRangeHelp'}
+              data-test-subj={'mlModelsStartDeploymentModalVCPULevel'}
             />
           </EuiFormRow>
-        </EuiDescribedFormGroup>
-      ) : null}
+
+          <EuiSpacer size={'s'} />
+
+          <EuiFormHelpText id={'vCpuRangeHelp'}>
+            <EuiCallOut size="s">
+              <p>{helperText}</p>
+            </EuiCallOut>
+          </EuiFormHelpText>
+        </EuiPanel>
+
+        {!disableAdaptiveResourcesControl ? (
+          <>
+            <EuiSpacer size={'m'} />
+
+            <EuiSwitch
+              label={
+                <FormattedMessage
+                  id="xpack.ml.trainedModels.modelsList.startDeployment.adaptiveAllocationEnabledLabel"
+                  defaultMessage="Adaptive resources (recommended)"
+                />
+              }
+              checked={!!config.adaptiveResources}
+              onChange={(event) => {
+                onConfigChange({
+                  ...config,
+                  adaptiveResources: event.target.checked,
+                });
+              }}
+              aria-describedby={'adaptiveResourcesHelp'}
+              data-test-subj={'mlModelsStartDeploymentModalAdaptiveResources'}
+            />
+
+            <EuiSpacer size={'s'} />
+
+            <EuiFormHelpText
+              id={'adaptiveResourcesHelp'}
+              data-test-subj={'mlModelsStartDeploymentModalVCPULevelHelpText'}
+            >
+              <FormattedMessage
+                id={'xpack.ml.trainedModels.modelsList.startDeployment.adaptiveResourcesHelp'}
+                defaultMessage={
+                  'Adjust resources to optimize for load and savings. If disabled, ELSER will not auto-scale.'
```

Comment:
Should ELSER be hardcoded in this message? For example, when deploying an e5 model, it still refers to ELSER. 

`If disabled, the deployment will not auto-scale.` perhaps?

Reply:
Fixed in de01fd8c6d6fcb09b8960f189f1cd856e717f480